FirebasePerformance/Sources/FPRClient.m

```diff
@@ -209,8 +209,8 @@ - (void)logNetworkTrace:(nonnull FPRNetworkTrace *)trace {
     return;
   }
   dispatch_group_async(self.eventsQueueGroup, self.eventsQueue, ^{
-    FPRMSGNetworkRequestMetric *networkRequestMetric = FPRGetNetworkRequestMetric(trace);
-    if (networkRequestMetric) {
+    if (trace.hasValidResponseCode) {
```

Comment:
Consider creating an API `-(BOOL) isValid` inside FPRNetworkTrace that can determine if the trace is valid. Checking on the responseCode, though fine for now, is not scalable/maintainable. The FPRNetworkTrace is the right object that can determine if the trace is valid/not.

Reply:
Great suggestion! Updated.